pkg/operator/controller/certificate/publish_ca.go

```diff
@@ -52,10 +56,35 @@ func (r *reconciler) ensureRouterCAConfigMap(secret *corev1.Secret, ingresses []
 	return nil
 }
 
-// desiredRouterCAConfigMap returns the desired router CA configmap.
-func desiredRouterCAConfigMap(secret *corev1.Secret, ingresses []operatorv1.IngressController) (*corev1.ConfigMap, error) {
-	if !shouldPublishRouterCA(ingresses) {
-		return nil, nil
+// desiredRouterCAConfigMap returns a configmap containing a single 'ca-bundle.crt'
+// key whose value is the concatenation of:
+//
+// 1. The default generated CA tls.crt data
+// 2. All distinct user-provided default certificate secret tls.crt data referenced
+//    by ingresscontrollers.
+func (r *reconciler) desiredRouterCAConfigMap(ingresses []operatorv1.IngressController) (*corev1.ConfigMap, error) {
+	caName := controller.RouterCASecretName(r.operatorNamespace)
+	secrets := map[string]types.NamespacedName{
+		caName.String(): caName,
+	}
+	for _, ingress := range ingresses {
+		if cert := ingress.Spec.DefaultCertificate; cert != nil {
+			name := types.NamespacedName{Namespace: "openshift-ingress", Name: cert.Name}
+			secrets[name.String()] = name
+		}
+	}
+
+	var certs [][]byte
+	for _, name := range secrets {
+		secret := &corev1.Secret{}
+		if err := r.client.Get(context.TODO(), name, secret); err != nil {
```

Comment:
Better to list secrets use that list and the list of ingresses to build a map; see how `desiredRouterCertsGlobalSecret` builds `ingressToSecret`: https://github.com/openshift/cluster-ingress-operator/blob/b5d4fb5c3fae2c72f79dbc598f81c2271cd693cd/pkg/operator/controller/certificate-publisher/publish_certs.go#L59-L92

What do you think of keeping the parameter for the CA secret and adding a parameter for the list of secrets so that `desiredRouterCAConfigMap` doesn't need `r`?